src/utils/attachments.js

```diff
@@ -0,0 +1,45 @@
+import { ROUTE_FILES } from './routingConstants'
+
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+export default class Attachments {
+  constructor({ provider, namespace, name, revision, path, row }) {
+    this.provider = provider
+    this.namespace = namespace
+    this.name = name
+    this.revision = revision
+    this.path = path
+    this.row = row
+  }
+
+  getFileAttachmentUrl() {
+    if (this.row.children) return
+    if (this.provider === 'github' && this.path) return this.getGitUrl()
+    else if (this.row.token) return this.getFileRoute(this.row.token)
+    else if (this.row.hashes) return this.getFileRoute(this.row.hashes.sha256)
+  }
+
+  static async fetchAttachmentWithToken(token) {
+    const clearlyDefinedContent = await fetch(this.getClearlyDefinedUrl(token))
+    if (clearlyDefinedContent.status === 200) return clearlyDefinedContent.text()
+    const swhContent = await fetch(this.getSwhUrl(token))
+    if (swhContent.status === 200) return swhContent.text()
```

Comment:
we need to track whether we got the content from swh or clearlydefined so we can attribute back to them correctly